packages/flutter/lib/src/material/switch.dart

```diff
@@ -954,4 +955,10 @@ class _SwitchPainter extends ToggleablePainter {
       _isPainting = false;
     }
   }
+
+  @override
+  void dispose() {
+    _cachedThumbPainter?.dispose();
```

Comment:
Should we null _cachedThubPainter and other internal fields so that holding a reference to a disposed _MaterialSwitch doesn't prevent the image from being GC'd?

Reply:
Good idea, will make that change.